app/controllers/dependencies_controller.rb

```diff
@@ -29,52 +21,26 @@ def edit
   def update
     @dependency = Dependency.find(params[:id])
 
-    respond_to do |format|
-      if @dependency.update_attributes(dependency_params)
-        format.html { redirect_to @dependency, :notice => 'Dependency was successfully updated.' }
-        format.json { head :ok }
-      else
-        format.html {
-          render :action => "edit"
-        }
-        format.json { render :json => @dependency.errors, :status => :unprocessable_entity }
-      end
-    end
+    flash[:notice] = 'Dependency was successfully updated' if @dependency.update_attributes(dependency_params)
+    respond_with(@dependency)
   end
 
   def new
     @dependency = Dependency.new
-
-    respond_to do |format|
-      format.html
-      format.json { render :json => @dependency }
-    end
   end
 
   def create
     @dependency = Dependency.new(dependency_params)
 
-    respond_to do |format|
-      if @dependency.save
```

Comment:
I think you removed the line that actually saves the record :) This old article has some good examples of applying respond_with to the create action: http://blog.plataformatec.com.br/2009/08/embracing-rest-with-mind-body-and-soul/